<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CppTools::Internal::CppCodeStyleSettingsPage</class>
 <widget class="QWidget" name="CppTools::Internal::CppCodeStyleSettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>757</width>
    <height>465</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QWidget" name="fallbackContainer" native="true">
     <layout class="QHBoxLayout" name="fallbackLayout">
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="TextEditor::FallbackSelectorWidget" name="fallbackWidget" native="true"/>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="categoryTab">
     <property name="currentIndex">
      <number>4</number>
     </property>
     <widget class="QWidget" name="generalTab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="TextEditor::TabPreferencesWidget" name="tabPreferencesWidget" native="true">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="TextEditor::SnippetEditorWidget" name="previewTextEditGeneral">
         <property name="plainText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="contentTab">
      <attribute name="title">
       <string>Content</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QGroupBox" name="contentGroupBox">
         <property name="title">
          <string>Indent</string>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="indentAccessSpecifiers">
            <property name="text">
             <string>&quot;public&quot;, &quot;protected&quot; and
&quot;private&quot; within class body</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentDeclarationsRelativeToAccessSpecifiers">
            <property name="text">
             <string>Declarations relative to &quot;public&quot;,
&quot;protected&quot; and &quot;private&quot;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentFunctionBody">
            <property name="text">
             <string>Statements within method body</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentBlockBody">
            <property name="text">
             <string>Statements within blocks</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentNamespaceBody">
            <property name="text">
             <string>Declarations within
&quot;namespace&quot; definition</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>17</width>
              <height>114</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="TextEditor::SnippetEditorWidget" name="previewTextEditContent">
         <property name="plainText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="bracesTab">
      <attribute name="title">
       <string>Braces</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QGroupBox" name="bracesGroupBox">
         <property name="title">
          <string>Indent Braces</string>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="indentClassBraces">
            <property name="text">
             <string>Class declarations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentNamespaceBraces">
            <property name="text">
             <string>Namespace declarations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentEnumBraces">
            <property name="text">
             <string>Enum declarations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentFunctionBraces">
            <property name="text">
             <string>Method declarations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentBlockBraces">
            <property name="text">
             <string>Blocks</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>195</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="TextEditor::SnippetEditorWidget" name="previewTextEditBraces">
         <property name="plainText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="switchTab">
      <attribute name="title">
       <string>&quot;switch&quot;</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QGroupBox" name="switchGroupBox">
         <property name="title">
          <string>Indent within &quot;switch&quot;</string>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="indentSwitchLabels">
            <property name="text">
             <string>&quot;case&quot; or &quot;default&quot;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentCaseStatements">
            <property name="text">
             <string>Statements relative to
&quot;case&quot; or &quot;default&quot;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentCaseBlocks">
            <property name="text">
             <string>Blocks relative to
&quot;case&quot; or &quot;default&quot;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="indentCaseBreak">
            <property name="text">
             <string>&quot;break&quot; statement relative to
&quot;case&quot; or &quot;default&quot;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>143</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="TextEditor::SnippetEditorWidget" name="previewTextEditSwitch">
         <property name="plainText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="alignmentTab">
      <attribute name="title">
       <string>Alignment</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QGroupBox" name="alignmentGroupBox">
         <property name="title">
          <string>Align</string>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QCheckBox" name="alignAssignments">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;
Enables alignment to tokens after =, += etc. When the option is disabled, regular continuation line indentation will be used.&lt;br&gt;
&lt;br&gt;
With alignment:
&lt;pre&gt;
a = a +
    b
&lt;/pre&gt;
Without alignment:
&lt;pre&gt;
a = a +
        b
&lt;/pre&gt;
&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Align after assignments</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="extraPaddingConditions">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;
The extra padding usually only affects if statement conditions. Without extra padding:
&lt;pre&gt;
if (a &amp;&amp;
    b)
    c;
&lt;/pre&gt;
With extra padding:
&lt;pre&gt;
if (a &amp;&amp;
        b)
    c;
&lt;/pre&gt;
&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Add extra padding to conditions
if they would align to the next line</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="TextEditor::SnippetEditorWidget" name="previewTextEditPadding">
         <property name="plainText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TextEditor::SnippetEditorWidget</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">texteditor/snippets/snippeteditor.h</header>
  </customwidget>
  <customwidget>
   <class>TextEditor::TabPreferencesWidget</class>
   <extends>QWidget</extends>
   <header location="global">texteditor/tabpreferenceswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TextEditor::FallbackSelectorWidget</class>
   <extends>QWidget</extends>
   <header location="global">texteditor/fallbackselectorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
